test/e2e/clusterctl_upgrade.go

```diff
@@ -71,6 +71,24 @@ type ClusterctlUpgradeSpecInput struct {
 	// InitWithKubernetesVersion can be used to override the INIT_WITH_KUBERNETES_VERSION e2e config variable with a specific
 	// Kubernetes version to use to create the secondary management cluster, e.g. `v1.25.0`
 	InitWithKubernetesVersion string
+	// InitWithCoreProvider can be used to override the INIT_WITH_CORE_PROVIDER e2e config variable with a specific core provider
+	// version to use when initializing the secondary management cluster, e.g. `cluster-api:v1.3.0`
```

Comment:
```suggestion
	// InitWithCoreProvider specifies the core provider version to use when initializing the secondary management cluster, e.g. `cluster-api:v1.3.0`.
	// If not set, the core provider version is calculated based on the contract.
```
nit. same below (there is no INIT_WITH_CORE_PROVIDER)

Reply:
Q: should we document these take precedence on InitWithProvidersContract? (same for the others)